packages/core/src/StateMachine.ts

```diff
@@ -66,9 +66,9 @@ export class StateMachine<
   TTag extends string,
   TInput,
   TOutput,
-  TEmitted extends EventObject = EventObject, // TODO: remove default
-  TMeta extends MetaObject = MetaObject,
-  TConfig extends StateSchema = StateSchema
+  TEmitted extends EventObject,
+  TMeta extends MetaObject,
+  TConfig extends StateSchema
```

Comment:
This is the main change. By having these as optional, we were (well, at least I was) miscounting the generic type params, and some were erroneously using their default types.